src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -11494,5 +11495,753 @@ .maxstack  0
+                // (8,31): error CS1746: The delegate '<anonymous delegate>' does not have a parameter named 'b'
+                //         Console.WriteLine(lam(b: "b", c: "a", a: "c"));
+                Diagnostic(ErrorCode.ERR_BadNamedArgumentForDelegateInvoke, "b").WithArguments("<anonymous delegate>", "b").WithLocation(8, 31));
+        }
+
+        [Fact]
+        public void LambdaOutOfOrderParameterInvocation_MissingOptionalParameter()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        var lam = (string a, string b, string c = "c") => $"{a}{b}{c}";
+        Console.WriteLine(lam(b: "a", a: "b"));
+    }
+}
+""";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,31): error CS1746: The delegate '<anonymous delegate>' does not have a parameter named 'b'
+                //         Console.WriteLine(lam(b: "a", a: "b"));
+                Diagnostic(ErrorCode.ERR_BadNamedArgumentForDelegateInvoke, "b").WithArguments("<anonymous delegate>", "b").WithLocation(8, 31));
+        }
+
+        [Fact]
+        public void LambdaOptionalParameterDecimalExpression()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        var lam = (decimal dec =  Decimal.One / (decimal) 3) => dec; 
+        Console.WriteLine(lam());
+    }
+
+}
+""";
+            CompileAndVerify(source, expectedOutput: "0.3333333333333333333333333333");
+        }
+
+
+        // PROTOTYPE: Do we want to allow [Caller{MemberName, LineNumber, FilePath, ArgumentExpression}] attributes for lambdas since
+        // we now have default parameters? The current behavior is to ignore these attributes so that the provided
+        // default would always be used in these cases.
+
+        [Fact]
+        public void CallerAttributesOnLambdaWithDefaultParam()
+        {
+            var source = """
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    public static void Main()
+    {
+        var lam = ([CallerMemberName] string member = "member", [CallerFilePath] string filePath = "file", [CallerLineNumber] int lineNumber = 0) => Console.WriteLine($"{filePath}::{member}:{lineNumber}");
+        lam();
+    }
+}
+""";
+            CompileAndVerify(source, expectedOutput: "file::member:0");
+        }
+
+
+        [Fact]
+        public void CallerArgumentExpressionAttributeOnLambdaWithDefaultParam()
+        {
+            var source = """
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    public static void Main()
+    {
+        var lam = (int arg, [CallerArgumentExpression("arg")] string argExpression = "callerArgExpression") => Console.WriteLine($"{argExpression}");
+        lam(3);
+    }
+}
+""";
+            CompileAndVerify(source, targetFramework: TargetFramework.Net60, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Passes : Verification.Skipped);
```

Comment:
>); [](http://example.com/codeflow?start=162&length=2)

Should this include `expectedOutput: "3"`?

Reply:
Expected output is actually the default value since the `[CallerArgumentExpression]` attribute isn't supported on lambda parameters. But yes, it does need expected output I just fixed that